server/aesdsocket.c

```diff
@@ -161,32 +187,27 @@ void receive_send_data(void* thread_param)
 	//data receive from client
 	do
 	{
-                received_bytes = recv(thread_func_args->client_fd, thread_func_args->buffer + buf_loc, MAXDATASIZE, 0); //sizeof(buffer)
+                received_bytes = recv(thread_func_args->client_fd, thread_func_args->buffer, MAXDATASIZE - 1, 0); //sizeof(buffer)
```

Comment:
@Sharath-Jonnala I think with this latest refactor to remove realloc there are some new assumptions about MAXDATASIZE.  I think it now is assumed this is > than the sum of bytes of all packets sent.  I'd expect this to cause problems.